### PR TITLE
feat: migrate user data to backend

### PR DESCRIPTION
**Pull request by imagine-hussain**

### CODE REVIEW

**Review by zax-xyz — APPROVED** [review]
didn't read but lgtm

**Comment by Peedee2002** [comment]
> didn't read but lgtm

image
mmhmm shippable code
yummy yummy raw html

**Comment by imagine-hussain** [comment]
> didn't read but lgtm

real

File: .github/workflows/ci_BE.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Backend Unit Tests
 
 on:
   pull_request:
-    branches: [dev, main]
+    branches: [dev, main, CIRCLES-331/user-data-to-backend]
```
Comment (lhjt) [inline .github/workflows/ci_BE.yml]: This (and other occurrences) need to be removed
```suggestion
    branches: [dev, main]
```

File: backend/server/config.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 URI = "mongodb://mongodb:27017/?readPreference=primary&appname=MongoDB%20Compass&directConnection=true&ssl=false"
 
+CLIENT_ID = "1017197944285-i4ov50aak72667j31tuieffd8o2vd5md.apps.googleusercontent.com"
```
Comment (lhjt) [inline backend/server/config.py:5]: This shouldn't be hardcoded and extracted to an environment variable

File: backend/server/database.py [hunk 3]
```diff
@@ -65,10 +69,138 @@ def overwrite_archives():
...
 
 def overwrite_all():
     """Singular execution point to overwrite the entire database including the archives"""
     overwrite_collection("Courses")
     overwrite_collection("Specialisations")
     overwrite_collection("Programs")
     overwrite_archives()
+
+def optionally_create_new_data():
+    # TODO: DELETE NEXT LINE IN PROD
```
Comment (lhjt) [inline backend/server/database.py]: (I assume this should be removed?)

File: backend/server/routers/auth.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+def validate_login(token) -> bool:
+    """
+        Take a token and validate a login off the following criteria
+            - token i
+            - token has not expired (given in unix time)
+            - email verified
+            - email exists in our BE
+    """
+    return token
```
Comment (lhjt) [inline backend/server/routers/auth.py:31]: Uh (?)

File: frontend/src/pages/Auth/Auth.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
+/* eslint-disable */
```
Comment (lhjt) [inline frontend/src/pages/Auth/Auth.tsx:1]: Why is this included?

File: frontend/src/pages/Auth/Auth.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    console.log("callBackResponse with jwt: ", jwt);
+    setUserObject(jwt_decode(jwt));
+
+    
+    return 0;
+  }
+
+  useEffect(() => {
+    (window as any).google.accounts.id.initialize({
+      client_id: "1017197944285-i4ov50aak72667j31tuieffd8o2vd5md.apps.googleusercontent.com",
```
Comment (lhjt) [inline frontend/src/pages/Auth/Auth.tsx:53]: Should be injected at build time via environment variable

**Review by lhjt — CHANGES_REQUESTED** [review]

**Comment by lhjt** [comment]
Oh rip this was marked as ready to review as I was reviewing it my bad

**Comment by Peedee2002** [comment]
sorry @lhjt i was trying to update the branch from my phone

File: backend/server/database.py [hunk 7]
```diff
@@ -65,10 +67,138 @@ def overwrite_archives():
             except (KeyError, IOError, OSError):
                 print(f"Failed to load and overwrite {year} archive")
 
+def create_dynamic_db():
+    usersDB.create_collection('users',
```
Comment (Peedee2002) [inline backend/server/database.py:71]: example validation

File: backend/server/routers/user.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        objectID = usersDB['users'].insert_one(dict(item)).inserted_id
+        usersDB['tokens'].insert_one({'token': token, 'objectId': objectID})
+
+
+@router.post("/saveLocalStorage/")
+def save_local_storage(localStorage: LocalStorage, token: str = DUMMY_TOKEN):
+    # TODO: turn giving no token into an error
+    # THINK: this is wierd security wise, because we are setting it to what we are given no ma
+    item = {
+        'degree': localStorage.degree,
```
Comment (Peedee2002) [inline backend/server/routers/user.py:49]: example insert

File: backend/server/routers/model.py [hunk 9]
```diff
@@ -153,6 +154,29 @@ class Config:
...
+
+class PlannerLocalStorage(TypedDict):
+    mostRecentPastTerm: MostRecentPastTerm
+    unplanned: list[str]
+    startYear: int
+    numYears: int
+    isSummerEnabled: bool
+    years: list[dict[str, None | list[str]]]
+
+class Storage(TypedDict):
```
Comment (Peedee2002) [inline backend/server/routers/model.py:199]: thing i am typechecking (?)

File: backend/server/routers/auth.py (hunk 4 above)
Comment (imagine-hussain) [reply backend/server/routers/auth.py:31]: TODO: not this

File: backend/server/config.py (hunk 2 above)
Comment (imagine-hussain) [reply backend/server/config.py:5]: agreed; will be done *eventually*

**Pull request closed without merging**